recipe_scrapers/donnahay.py

```diff
@@ -0,0 +1,131 @@
+# mypy: allow-untyped-defs
+
```

Comment:
This pull request is generally ready I think - just some merge conflicts to resolve.

There's a small cleanup opportunity here too - after #1174 we don't need these `allow-untyped-defs` `mypy` directives, so this can be removed from the file header.